### ISSUE TITLE
darknet_video.py output an empty video file.

### ISSUE THREAD

**Issue by AbdulmalikAlyousfi** [AlexeyAB/darknet#7753]
when running darknet_video.py using the following command:

python3 darknet_video.py --input test.mp4 --out_filename data/test.mp4 --weights backup/yolov3-custom_best.weights --dont_show --thresh 0.5

The output is an empty file with a size of 258 byte if the format is .mp4 or 5.6k byte if the format is .avi

**Comment by johncch** [comment]
I had the same issue and what solved it for me was redefining the width and height of the video file. The width and height that were derived from the darknet network parameters were different from the video frames fed to it, and as a result the video writer wouldn't write any frames.

**Comment by kndt84** [comment]
If your OpenCV version is 4.X, darknet does not write out movie files correctly. If possible, try to use OpneCV 3.X.

**Comment by stephanecharette** [comment]
> If your OpenCV version is 4.X, darknet does not write out movie files correctly. If possible, try to use OpneCV 3.X.

Where did you get that?  I've been using OpenCV v4.2 without any problems.

**Comment by kndt84** [comment]
I got it from Ubuntu20.04 apt repository with the following command
`apt install libopencv-dev`

**Comment by stephanecharette** [comment]
No, the "where did you get that" was about the false comment that darknet does not write out movie files when using OpenCV 4.x.

**Comment by LAsurvivor** [comment]
> I had the same issue and what solved it for me was redefining the width and height of the video file. The width and height that were derived from the darknet network parameters were different from the video frames fed to it, and as a result the video writer wouldn't write any frames.

Can u plz tell how to redefine the width and the heitght of the video file?

**Comment by johncch** [comment]
I basically rewrote lines 176 and 177 and hard coded it with the values from my input file (that I got from the file info).

<img width="963" alt="image" src="https://user-images.githubusercontent.com/331432/124792908-51c92f00-df02-11eb-9105-843fe85ce486.png">

**Comment by johncch** [comment]
To expand on this a bit: 
`network_width` and `network_height` was 512 on the computer I was using when I printed it out.
I'm not familiar with the internals of darknet, but I assume it refers to [this in the documentation](https://github.com/AlexeyAB/darknet#geforce-rtx-2080-ti)
My input file was 1920 by 1080. So when I rewrote it to width 1920 and height 1080 the file was written correctly.

I do not have a generalized solution which is why I did not submit a patch

**Comment by SimrnGupta** [comment]
> To expand on this a bit:
> `network_width` and `network_height` was 512 on the computer I was using when I printed it out.
> I'm not familiar with the internals of darknet, but I assume it refers to [this in the documentation](https://github.com/AlexeyAB/darknet#geforce-rtx-2080-ti)
> My input file was 1920 by 1080. So when I rewrote it to width 1920 and height 1080 the file was written correctly.
> 
> I do not have a generalized solution which is why I did not submit a patch

![Screenshot 2021-07-09 at 8 56 37 PM](https://user-images.githubusercontent.com/54317029/125101745-2f990200-e0f8-11eb-89c5-72961d757f97.png)

I added this line to the drawing function, so the video writer writes the frames.
Though the output file is resized to the darknet network's dimensions on lines 176 and 177

**Comment by LAsurvivor** [comment]
> To expand on this a bit:
> `network_width` and `network_height` was 512 on the computer I was using when I printed it out.
> I'm not familiar with the internals of darknet, but I assume it refers to [this in the documentation](https://github.com/AlexeyAB/darknet#geforce-rtx-2080-ti)
> My input file was 1920 by 1080. So when I rewrote it to width 1920 and height 1080 the file was written correctly.
> 
> I do not have a generalized solution which is why I did not submit a patch

What file it that? Could you show me the lines after changed?

**Comment by LAsurvivor** [comment]
I used the command in this [repo](https://github.com/CityofEdmonton/AI-Pothole-Detection-Using-YOLO/blob/master/documentation/RUNNING_THE_PROJECT.md)
`./darknet detector demo cfg/obj.data cfg/yolo-pothole-test.cfg <trained_weight_file> <video_file> -out_filename <video_output>.avi
`
But the output video only shows several frames and I don't know why

**Comment by LAsurvivor** [comment]
> when running darknet_video.py using the following command:
> 
> python3 darknet_video.py --input test.mp4 --out_filename data/test.mp4 --weights backup/yolov3-custom_best.weights --dont_show --thresh 0.5
> 
> The output is an empty file with a size of 258 byte if the format is .mp4 or 5.6k byte if the format is .avi

When I tried this command, it shows that
```
OSError: /home/usr/darknet/libdarknet.so: cannot open shared object file: No such file or directory
```

**Comment by SimrnGupta** [comment]
> > when running darknet_video.py using the following command:
> > python3 darknet_video.py --input test.mp4 --out_filename data/test.mp4 --weights backup/yolov3-custom_best.weights --dont_show --thresh 0.5
> > The output is an empty file with a size of 258 byte if the format is .mp4 or 5.6k byte if the format is .avi
> 
> When I tried this command, it shows that
> 
> ```
> OSError: /home/usr/darknet/libdarknet.so: cannot open shared object file: No such file or directory
> ```

In the makefile, enable LIBSO, i.e., change the value to LIBSO=1 before running the make command.

I used this code :
```
# change makefile to have GPU, OPENCV and LIBSO enabled
%cd darknet
!sed -i 's/OPENCV=0/OPENCV=1/' Makefile
!sed -i 's/GPU=0/GPU=1/' Makefile
!sed -i 's/CUDNN=0/CUDNN=1/' Makefile
!sed -i 's/CUDNN_HALF=0/CUDNN_HALF=1/' Makefile
!sed -i 's/LIBSO=0/LIBSO=1/' Makefile
```
